pkg/argocd/update.go

```diff
@@ -408,27 +411,66 @@ func marshalParamsOverride(app *v1alpha1.Application, originalData []byte) ([]by
 		mergeKustomizeOverride(&params, &newParams)
 		override, err = yaml.Marshal(params)
 	case ApplicationTypeHelm:
-		if app.Spec.Source.Helm == nil {
+		if appSource.Helm == nil {
 			return []byte{}, nil
 		}
-		var params helmOverride
-		newParams := helmOverride{
-			Helm: helmParameters{
-				Parameters: app.Spec.Source.Helm.Parameters,
-			},
-		}
 
-		if len(originalData) == 0 {
-			override, err = yaml.Marshal(newParams)
-			break
-		}
-		err = yaml.Unmarshal(originalData, &params)
-		if err != nil {
-			override, err = yaml.Marshal(newParams)
-			break
+		if strings.HasPrefix(app.Annotations[common.WriteBackTargetAnnotation], common.HelmPrefix) {
+			images := GetImagesFromApplication(app)
+
+			for _, c := range images {
+				helmAnnotationParamName, helmAnnotationParamVersion := getHelmParamNamesFromAnnotation(app.Annotations, c.ImageName)
+				if helmAnnotationParamName == "" {
+					return nil, fmt.Errorf("could not find an image-name annotation for image %s", c.ImageName)
```

Comment:
I mean's,that c.ImageName is mistake,not use image-list defined alias,may be use c.ImageAlias is right?

Reply:
I review more code,I found if you use like ContainerImage's normalizedSymbolicNam is good idea,If there are multiple layers of "/" in image, it is not legal to use the annotation name.There's no image list in this place?

Reply:
If i don't set helmvalues in write-back-target then it is able to commit, it finds proper annotations and etc. But as soon as I set write-back-target and specify values.yaml it is failing with error: "Could not update application spec: could not find an image-name annotation for". Should we register a bug for that usecase?

Reply:
@pavel-kurnosov , this is the same as @SamPeng87 noticed. I'll take a look into it when I have some more time.
If you want to open an issue I think it's ok, then we'll be able to reference it when pushing the PR to solve this.

Thank you!